com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1175,6 +1177,11 @@ private void OnNetworkPreUpdate()
         private void OnNetworkPostLateUpdate()
         {
             m_MessagingSystem.ProcessSendQueues();
+            if (m_LastMetricsTickId != NetworkTickSystem.LocalTime.Tick)
```

Comment:
As this is called after late update I'm guessing this works to make sure that we're not dispatching information in the same frame twice, what is slightly confusing is that we're using tick to determine this, but we're not dispatching on network tick any more. I think it might be worth leaving a comment just to make sure that is clear and this isn't misread?

Reply:
Your comment made me realize that this implementation was faulty. What it was doing was making sure that metrics only dispatch on frames where a tick occurs. However, this is wrong! Some of our metrics are tracked synchronously when users take an action and aren't deferred until the next tick.

I changed the implementation to instead use a dirty flag in NetworkMetrics. This way, metrics are only sent if metrics are recorded, regardless of whether a tick occurs.

I added/updated tests to reflect this.